server/navigation/nav_dict.py

```diff
@@ -0,0 +1,63 @@
+import _manual_routes, _rules
+
+_nav_dict = {}  # ONLY ACCESS IN THIS MODULE
+
+def create_dict(options_list):
+    length_before_docs = 0
+
+    if len(_nav_dict) == 0:
+        _nav_dict.update(_manual_routes.get_manual_routes())
+        length_before_docs = len(_nav_dict)
+
+    if len(_nav_dict) > 0:
+        rules_dict = _rules.generate_all_routes_dict(options_list)
+
+        if rules_dict != None and len(rules_dict) > 0:
+            _nav_dict.update(rules_dict)
+
+        if len(_nav_dict) == length_before_docs:
+            print("Looked for documents, but didn't find any. See config/README.md to configure rules docs.")
+
+    else:
+        print("Error: _nav_dict not created.")
```

Comment:
I would kinda like to see error handling logged, but asking that would scope creep.

Reply:
Yeah, I'm wary about passing the logger to everything, but that may be the Python way? I could add it if you want. I was thinking of it, but decided to submit what I had so far as it could be addressed later.